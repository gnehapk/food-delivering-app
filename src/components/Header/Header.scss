@import  '../../color-variables.scss';

.header {
  display: flex;
  background-color: $color-egg-blue;
  padding: 1.5rem 4rem;
  flex-direction: column;
  justify-content: flex-start;

  @media only screen and (min-width: 768px) {
    flex-direction: row;
    justify-content: space-between;
  }
  

  .location-name {
    color: $color-white;
    font-size: 1.5rem;
    font-weight: bold;
  }

  .location-text {
    margin-bottom: 0;
  }

  .btn-container {
    display: flex;
    align-items: center;
    
    .location-btn {
      color: $color-egg-blue;
      height: 40px;
    }
  }
}