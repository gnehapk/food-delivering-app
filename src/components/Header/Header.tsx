import React from 'react';
import Button from 'react-bootstrap/Button';
import './Header.scss';

const Header: React.FC<HeaderProps> = (props) => (
  <div className="header">
    <div>
      <p className="location-text">Location</p>
      <p className="location-name">{props.location}</p>
    </div>
    <div className="btn-container">
      <Button variant="light" className="location-btn">Change Location</Button>
    </div>
  </div>
);

type HeaderProps = {
  location: string;
};

export default Header;