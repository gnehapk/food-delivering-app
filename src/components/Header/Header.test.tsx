import React from 'react';
import TestRenderer from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(<Header location="Tokyp" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

