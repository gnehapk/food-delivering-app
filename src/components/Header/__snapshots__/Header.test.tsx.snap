// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="header"
>
  <div>
    <p
      className="location-text"
    >
      Location
    </p>
    <p
      className="location-name"
    >
      Tokyp
    </p>
  </div>
  <div>
    <button
      className="location-btn btn btn-light"
      disabled={false}
      type="button"
    >
      Change Location
    </button>
  </div>
</div>
`;
